packages/monaco/src/browser/monaco-outline-contribution.ts

```diff
@@ -64,6 +68,38 @@ export class MonacoOutlineContribution implements FrontendApplicationContributio
                 });
             }
         });
+
+        this.editorManager.onCurrentEditorChanged(widget => {
```

Comment:
it should be a responsibility of the editor extension

Reply:
I agree with you @akosyakov . My bad. 

I moved this part to `editor-contributions.ts`. As you might have noticed, `packages/editor/src/typings/index.d.ts` was also added. The reason behind it was, without the new `index.d.ts`, I am unable to call `getOptions()` to access the monaco model. I am not sure if this is against the best practices. Please advise.

Reply:
In the editor extension, we have an abstract editor which shields us from changes in Monaco. But obviously, it is not powerful enough. If you want to make this change to work not only for Monaco when you should program against the abstract editor and expose necessary Monaco apis: https://github.com/theia-ide/theia/pull/1512#discussion_r175021336. If you are fine to make it work only for Monaco then you should implement in the monaco extension.

Reply:
Yea I was actually wondering about this also. We've discussed it offline and we didn't know if this api should be monaco specific or really generic...

Reply:
Yes. I am using the editor package as an interface of the monaco. That's why I added `getModel(): editor.IModel;` to the `editor.ts` to shield the `getModel()` in monaco. 

If some day in future, another editor, say XYZ, is added as an extension, we simply need to do 
`class XYZEditor implements TextEditor { getModel(): editor.IModel}`.

Reply:
Right at the moment, there is only one function (i.e., getOptions()) exposed in the editor.IModel in packages/editor/src/typings/index.d.ts. Maybe that's what you are disagreeing @akosyakov  ?

For now that getOptions() is the only thing I need. Are you saying I should either expose more functions in the editor.IModel, or remove the editor.IModel & do everything in monaco.contributions (instead of editor.contributions)?

Reply:
> Yea I was actually wondering about this also. We've discussed it offline and we didn't know if this api should be monaco specific or really generic...

ok, let's just do it in the monaco extension, don't expose anything it the editor extension